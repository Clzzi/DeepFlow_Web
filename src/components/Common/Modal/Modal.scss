.Modal {
	&-Overlay {
		position: fixed;
		top: 0;
		left: 0;
		right: 0;
		bottom: 0;
		background-color: rgba(0, 0, 0, 0.4);
		z-index: 2;
	}

	&-Box {
		position: fixed;
		top: 50%;
		left: 50%;
		transform: translate(-50%, -50%);
		z-index: 3;
		background-color: var(--theme);
		border-radius: 5px;
		padding: 1rem;
		overflow-x: hidden;
		overflow-y: auto;

		&-Top {
			display: flex;
			display: -webkit-flex;
			justify-content: space-between;
			align-items: center;
			border-bottom: 0.5px solid var(--gray);
			padding-bottom: 0.4rem;
			margin-bottom: 0.5rem;

			&-Title {
				font-size: var(--font-small);
				font-weight: 500;
			}

			&-Close {
				font-size: var(--font-normal);
				cursor: pointer;
				transition: 0.15s ease-in-out;

				&:hover {
					color: var(--red);
				}
			}
		}
	}
}
