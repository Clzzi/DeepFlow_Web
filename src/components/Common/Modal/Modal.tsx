import { ReactNode } from 'react';
import classNames from 'classnames';
import { ClassNamesFn } from 'classnames/types';
import { VscChromeClose } from 'react-icons/vsc';

const style = require('./Modal.scss');
const cx: ClassNamesFn = classNames.bind(style);

interface ModalProps {
  title: string;
  onChangeIsModal: () => void;
  children?: ReactNode;
}

const Modal = ({
  title,
  onChangeIsModal,
  children,
}: ModalProps): JSX.Element => {
  return (
    <div className={cx('Modal')}>
      <div className={cx('Modal-Overlay')} onClick={onChangeIsModal}></div>

      <div className={cx('Modal-Box')}>
        <div className={cx('Modal-Box-Top')}>
          <div className={cx('Modal-Box-Top-Title')}>{title}</div>
          <VscChromeClose
            className={cx('Modal-Box-Top-Close')}
            onClick={onChangeIsModal}
          />
        </div>

        {children && children}
      </div>
    </div>
  );
};

export default Modal;
