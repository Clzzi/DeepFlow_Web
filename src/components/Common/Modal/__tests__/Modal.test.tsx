import { fireEvent, render } from '@testing-library/react';
import Modal, { ModalProps } from '../Modal';

describe('Modal', () => {
  const initialSetup = (props: Partial<ModalProps> = {}) => {
    const initialProps: ModalProps = {
      width: '100px',
      height: '100px',
      title: '제목',
      onChangeIsModal: () => {},
      ...props,
    };

    const utils = render(<Modal {...initialProps} />);
    const modalClose: HTMLElement[] = utils.getAllByTestId('modal-close');

    return {
      ...utils,
      modalClose,
    };
  };

  it('render correctly', () => {
    const { container } = initialSetup();
    expect(container).toMatchSnapshot();
  });

  it('well handling click', () => {
    const { modalClose } = initialSetup();
    
    for (const close of modalClose) {
      fireEvent.click(close);
    }
  });
});