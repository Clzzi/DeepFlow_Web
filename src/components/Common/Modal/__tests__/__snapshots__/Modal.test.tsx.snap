// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal render correctly 1`] = `
<div>
  <div
    class="Modal"
    data-testid="modal"
  >
    <div
      class="Modal-Overlay"
      data-testid="modal-close"
    />
    <div
      class="Modal-Box"
      style="width: 100px; height: 100px;"
    >
      <div
        class="Modal-Box-Top"
      >
        <div
          class="Modal-Box-Top-Title"
        >
          제목
        </div>
        <svg
          class="Modal-Box-Top-Close"
          data-testid="modal-close"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M7.116 8l-4.558 4.558.884.884L8 8.884l4.558 4.558.884-.884L8.884 8l4.558-4.558-.884-.884L8 7.116 3.442 2.558l-.884.884L7.116 8z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
